IPython/core/ultratb.py

```diff
@@ -248,6 +248,7 @@ def fix_frame_records_filenames(records):
         # Look inside the frame's globals dictionary for __file__,
         # which should be better. However, keep Cython filenames since
         # we prefer the source filenames over the compiled .so file.
+        filename = py3compat.cast_unicode(filename, "utf-8")
```

Comment:
Do I need to do it just for the calls I introduced for all calls to `cast_unicode` in this file